rclcpp/include/rclcpp/experimental/intra_process_manager.hpp

```diff
@@ -361,13 +361,16 @@ class IntraProcessManager
       if (subscription_it == subscriptions_.end()) {
         throw std::runtime_error("subscription has unexpectedly gone out of scope");
       }
-      auto subscription_base = subscription_it->second.subscription;
+      auto subscription_base_weak = subscription_it->second.subscription;
+      if (auto subscription_base = subscription_base_weak.lock()) {
```

Comment:
```suggestion
      auto subscription_base = subscription_it->second.subscription.lock();
      if (subscription_base) {
```

sounds a bit more clear.